dashboard/lib/metrics/events.rb

```diff
@@ -46,11 +46,34 @@ def log_event(user: nil, event_name:, event_value: nil, metadata: {}, get_enable
       )
       end
 
+      def log_event_with_session(session:, event_name:, event_value: nil, metadata: {})
+        event_value = event_name if event_value.nil?
+        statsig_user = StatsigUser.new({'userID' => session[:statsig_stable_id]})
+
+        if CDO.rack_env?(:development)
+          log_event_to_stdout_with_session(session: session, event_name: event_name, event_value: event_value, metadata: metadata)
+        elsif CDO.rack_env?(:production) || managed_test_environment
```

Comment:
You'll need to add `managed_test_environment = CDO.running_web_application? && CDO.test_system?` in this method similar to `log_event`.

Think it would be worth abstracting this and the `event_value = event_name if event_value.nil?` into two private methods that could be called in each of the log functions?

Reply:
I was thinking this would be a good candidate to either use a composition pattern or inheritance pattern. However I wanted to make sure the whole setup works well (with data flowing) before coming back and refactoring it. Thanks for catching this, I updated it :)

Reply:
Sounds good, yeah there's some opportunity to make the code a bit more dry, but ultimately I appreciate the explicit nature of the two different log methods. As you said, we can address a refactor/cleanup later once we verify it's working.